avocado/core/parser.py

```diff
@@ -80,4 +81,16 @@ def finish(self):
 
         Side effect: set the final value for attribute `args`.
         """
-        self.args = self.application.parse_args(namespace=self.args)
+        self.args, _ = self.application.parse_known_args(namespace=self.args)
+        if _:
+            for sub in self.application._subparsers._actions:
+                if sub.dest == 'subcommand':
+                    sub.choices[self.args.subcommand].print_usage()
+                    # Respecting the argparse exit code on unknown arg.
+                    # Using exit_codes.AVOCADO_JOB_FAIL here seems just wrong.
+                    sys.exit(2)
```

Comment:
@apahim it looks like it's intended to be configurable by overriding the `error` method itself. The docs read:

```
        error(message: string)

        Prints a usage message incorporating the message to stderr and
        exits.

        If you override this in a subclass, it should not return -- it
        should either exit or raise an exception.
```

Having said that, and understanding/agreeing that we should have consistency, we could:

1. Have our own `ArgumentParser` that exits with `exit_codes.AVOCADO_FAIL`
2. Swap the values of `exit_codes.AVOCADO_FAIL` and `exit_codes.AVOCADO_JOB_FAIL` so that they match argparse's own. Our own tests use the symbolic names, but *maybe* this can hit users relying on the current error codes.

In any case, I'd add a functional test that assures that `avocado bla` and `avocado run --bla` return the same error code, as it seems to be one of your goals.